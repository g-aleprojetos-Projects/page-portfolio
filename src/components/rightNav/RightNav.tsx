import React, {useState} from 'react';
import {useResponsiveContext} from 'context/mobileWindow';
import {useBackgroundContext} from 'context/backgroud';
import {ToggleSwitch} from 'components/toggleSwitch';
import * as S from './RightNav.styles';
import {TextoMedium, TextoSmall} from 'components/texto';
import {useNavigate} from 'react-router-dom';
import rotas from 'resourses/rotas';

type Props = S.PropsRightNav & {
  mudarEstado: () => void;
};

export const RightNav = (props: Props) => {
  const {openRightNav, mudarEstado} = props;
  const navigate = useNavigate();
  const {mobile} = useResponsiveContext();
  const {background, handleToggle} = useBackgroundContext();
  const [telaAtiva, setTelaAtiva] = useState<string>(rotas.Home);

  const handleNavegate = (item: string) => {
    setTelaAtiva(item);
    navigate(item);
    mudarEstado();
  };

  return (
    <>
      {mobile && openRightNav && <S.Container onClick={mudarEstado} />}
      <S.ContainerNav data-testid={'test_rightNav'}>
        <S.ContainerContent
          data-testid={'test_containerContent'}
          openRightNav={openRightNav}
          mobile={mobile}
          backgroundDark={background}>
          <S.Menu data-testid={'test_menu'} mobile={mobile}>
            <S.Texto>
              <TextoSmall>Menu</TextoSmall>
            </S.Texto>
          </S.Menu>
          <S.ItemNav
            mobile={mobile}
            isActive={telaAtiva === rotas.Home && true}
            onClick={() => {
              handleNavegate(rotas.Home);
            }}>
            {telaAtiva === rotas.Home ? (
              <TextoMedium peso="bold" cursor="default">
                Home
              </TextoMedium>
            ) : (
              <TextoSmall cursor="pointer"> Home</TextoSmall>
            )}
          </S.ItemNav>
          <S.ItemNav
            mobile={mobile}
            isActive={telaAtiva === rotas.Skills && true}
            onClick={() => {
              handleNavegate(rotas.Skills);
            }}>
            {telaAtiva === rotas.Skills ? (
              <TextoMedium peso="bold" cursor="default">
                Skills
              </TextoMedium>
            ) : (
              <TextoSmall cursor="pointer"> Skills</TextoSmall>
            )}
          </S.ItemNav>
          <S.ItemNav
            mobile={mobile}
            isActive={telaAtiva === rotas.Projetos && true}
            onClick={() => {
              handleNavegate(rotas.Projetos);
            }}>
            {telaAtiva === rotas.Projetos ? (
              <TextoMedium peso="bold" cursor="default">
                Projetos
              </TextoMedium>
            ) : (
              <TextoSmall cursor="pointer"> Projetos</TextoSmall>
            )}
          </S.ItemNav>
          <S.ItemNav
            mobile={mobile}
            isActive={telaAtiva === rotas.Sobre && true}
            onClick={() => {
              handleNavegate(rotas.Sobre);
            }}>
            {telaAtiva === rotas.Sobre ? (
              <TextoMedium peso="bold" cursor="default">
                Sobre
              </TextoMedium>
            ) : (
              <TextoSmall cursor="pointer"> Sobre</TextoSmall>
            )}
          </S.ItemNav>
          <S.Item data-testid={'test_toggle'}>
            <ToggleSwitch isOn={background} handleOnclick={handleToggle} />
          </S.Item>
          <S.Flag>
            <S.FlagBrasil data-testid={'test_flag_Brasil'} />
          </S.Flag>
        </S.ContainerContent>
      </S.ContainerNav>
    </>
  );
};
