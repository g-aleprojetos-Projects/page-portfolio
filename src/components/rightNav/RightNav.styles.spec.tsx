import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './RightNav.styles';

describe('RightNav.styles', () => {
  test('ContainerNav DEVE ser igual ao snapshot', () => {
    const containerNav = renderer.create(<S.ContainerNav />).toJSON();
    expect(containerNav).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "false" e backgroundDark for "false" ', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={false}
          mobile={false}
          backgroundDark={false}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "false" e backgroundDark for "true"', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={false}
          mobile={false}
          backgroundDark={true}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "true" e backgroundDark for "false"', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={false}
          mobile={true}
          backgroundDark={false}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "true" e for backgroundDark for "true" ', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={false}
          mobile={true}
          backgroundDark={true}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "false" e for backgroundDark for "false" ', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={true}
          mobile={false}
          backgroundDark={false}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "false" e for backgroundDark for "true" ', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={true}
          mobile={false}
          backgroundDark={true}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "true" e for backgroundDark for "false" ', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={true}
          mobile={true}
          backgroundDark={false}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "true" e for backgroundDark for "true" ', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent
          openRightNav={true}
          mobile={true}
          backgroundDark={true}
        />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });

  test('Texto DEVE ser igual ao snapshot', () => {
    const texto = renderer.create(<S.Texto />).toJSON();
    expect(texto).toMatchSnapshot();
  });
  test('Menu DEVE ser igual ao snapshot se mobile for igual a "true"', () => {
    const menu = renderer.create(<S.Menu mobile={true} />).toJSON();
    expect(menu).toMatchSnapshot();
  });
  test('Menu DEVE ser igual ao snapshot se mobile for igual a "false"', () => {
    const menu = renderer.create(<S.Menu mobile={false} />).toJSON();
    expect(menu).toMatchSnapshot();
  });
  test('Item DEVE ser igual ao snapshot', () => {
    const item = renderer.create(<S.Item />).toJSON();
    expect(item).toMatchSnapshot();
  });
  test('Flag DEVE ser igual ao snapshot', () => {
    const flag = renderer.create(<S.Flag />).toJSON();
    expect(flag).toMatchSnapshot();
  });
  test('FlagBrasil DEVE ser igual ao snapshot', () => {
    const flagBrasil = renderer.create(<S.FlagBrasil />).toJSON();
    expect(flagBrasil).toMatchSnapshot();
  });
});
