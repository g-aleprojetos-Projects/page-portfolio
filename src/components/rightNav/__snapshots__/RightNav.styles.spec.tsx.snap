// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "false" e backgroundDark for "false"  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: transparent;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  top: auto;
  right: auto;
  width: auto;
  height: auto;
  padding-top: auto;
  -webkit-transition: none;
  transition: none;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "false" e backgroundDark for "true" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: transparent;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  top: auto;
  right: auto;
  width: auto;
  height: auto;
  padding-top: auto;
  -webkit-transition: none;
  transition: none;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "true" e backgroundDark for "false" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  position: fixed;
  background-color: #000000;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  top: 0;
  right: 0;
  width: 100vw;
  height: auto;
  padding-top: 3.5rem;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "false", mobile for igual a "true" e for backgroundDark for "true"  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  position: fixed;
  background-color: #000000;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  top: 0;
  right: 0;
  width: 100vw;
  height: auto;
  padding-top: 3.5rem;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "false" e for backgroundDark for "false"  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: transparent;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  top: auto;
  right: auto;
  width: auto;
  height: auto;
  padding-top: auto;
  -webkit-transition: none;
  transition: none;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "false" e for backgroundDark for "true"  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: transparent;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  top: auto;
  right: auto;
  width: auto;
  height: auto;
  padding-top: auto;
  -webkit-transition: none;
  transition: none;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "true" e for backgroundDark for "false"  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  position: fixed;
  background-color: #000000;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  top: 0;
  right: 0;
  width: 100vw;
  height: auto;
  padding-top: 3.5rem;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot com openRightNav passado como "true", mobile for igual a "true" e for backgroundDark for "true"  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  position: fixed;
  background-color: #000000;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  top: 0;
  right: 0;
  width: 100vw;
  height: auto;
  padding-top: 3.5rem;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerNav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5.563rem;
  padding-right: 3.438rem;
}

<nav
  className="c0"
/>
`;

exports[`RightNav.styles Flag DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`RightNav.styles FlagBrasil DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 3.438rem;
  margin: 0.125rem;
}

<img
  alt="Bandeira do Brasil"
  className="c0"
  src="flagBrasil.svg"
/>
`;

exports[`RightNav.styles Item DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

<li
  className="c0"
/>
`;

exports[`RightNav.styles ItemNav DEVE ser igual ao snapshot quando "isActive" for "false" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

<li
  className="c0"
/>
`;

exports[`RightNav.styles ItemNav DEVE ser igual ao snapshot quando "isActive" for "true" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

<li
  className="c0"
/>
`;

exports[`RightNav.styles Menu DEVE ser igual ao snapshot se mobile for igual a "false" 1`] = `
.c0 {
  display: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 20px 0;
  margin-bottom: 10px;
  background-color: #00875F;
}

<div
  className="c0"
/>
`;

exports[`RightNav.styles Menu DEVE ser igual ao snapshot se mobile for igual a "true" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 20px 0;
  margin-bottom: 10px;
  background-color: #00875F;
}

<div
  className="c0"
/>
`;

exports[`RightNav.styles Texto DEVE ser igual ao snapshot 1`] = `
.c0 {
  color: #ffffff;
}

<div
  className="c0"
/>
`;
